internal/lifecycle/lifecycle.go

```diff
@@ -121,14 +149,29 @@ func (l *Lifecycle) Stop(ctx context.Context) error {
 		begin := time.Now()
 		if err := hook.OnStop(ctx); err != nil {
 			// For best-effort cleanup, keep going after errors.
+			l.logger.LogEvent(&fxevent.LifecycleHookExecuted{
+				FunctionName: funcName,
+				CallerName:   hook.callerFrame.Function,
+				Method:       _hookStop,
+				Err:          err,
+			})
 			errs = append(errs, err)
 		}
 		l.mu.Lock()
+		runtime := time.Since(begin)
 		l.stopRecords = append(l.stopRecords, HookRecord{
 			CallerFrame: hook.callerFrame,
 			Func:        hook.OnStop,
 			Runtime:     time.Since(begin),
 		})
+
+		l.logger.LogEvent(&fxevent.LifecycleHookExecuted{
```

Comment:
isn't it a bit confusing if you get two similar events, one on 152 and almost exact same one here w/out the Err but with timing into which you can get on 152 anyway? difference here from `OnStart` is that there we return the error right away but here we accumulate.